src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,28 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_COUNT_SYMBOL = 6;
+    public static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new CustomException("The user is not registered in the network");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new CustomException("User with this login already exists");
+        }
+        if (user.getLogin().length() < MIN_COUNT_SYMBOL) {
+            throw new CustomException("Is not valid login");
+        }
+        if (user.getPassword().length() < MIN_COUNT_SYMBOL) {
```

Comment:
The password length validation should check for null before calling the .length() method to avoid a potential NullPointerException.